pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestStart failed

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11600568?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11600568?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 1188523 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x63e1430, 0xc000134008}, {0xc001662cc8, 0x2, 0xc00d6b6200?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0x58
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2263 +0x105
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc00929cb60)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc00929cb60, 0xc00cc1c7e0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart2030293374
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31185

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11600569?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11600569?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 6332 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x68
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0xb046590, 0xc00013e018}, {0xc0049eb640, 0x2, 0xc003d70600?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0xd9
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x1f1
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2263 +0x16a
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0x134
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x158
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x93
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xf3
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc002ffaea0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xf5
testing.tRunner(0xc002ffaea0, 0xc005991880)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart3120709468
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11609734?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11609734?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 1180155 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x63e1430, 0xc000134008}, {0xc00147acc8, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0x58
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2263 +0x105
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc0073351e0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc0073351e0, 0xc004077540)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart3906847601
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11609735?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11609735?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 6218 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x68
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0xb046590, 0xc0001b4018}, {0xc0046db640, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0xd9
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x1f1
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2263 +0x16a
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0x134
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x158
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x93
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xf3
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc003d0a340)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xf5
testing.tRunner(0xc003d0a340, 0xc005c417e0)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart305279815
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11619752?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11619752?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 817188 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x63e1430, 0xc000084080}, {0xc0079c4cc8, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0x58
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2263 +0x105
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc00721b6c0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc00721b6c0, 0xc0023b41c0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart3877169333
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11630857?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11630857?buildTab=artifacts#/) on master @ [2b71df6ef0b1b86767b1e63b4071fbdf933e3b60](https://github.com/cockroachdb/cockroach/commits/2b71df6ef0b1b86767b1e63b4071fbdf933e3b60):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 1178053 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x63e1450, 0xc000084080}, {0xc0028facc8, 0x2, 0x5c5b53434a4f5200?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0x58
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2264 +0x105
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc0091db040)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc0091db040, 0xc00ba58fa0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart2659766373
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11649394?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11649394?buildTab=artifacts#/) on master @ [ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a](https://github.com/cockroachdb/cockroach/commits/ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 1172172 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x63fd8d0, 0xc0001ac008}, {0xc003550cc8, 0x2, 0xc0001ec000?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0x58
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2264 +0x105
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc00c3ce9c0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc00c3ce9c0, 0xc00db14de0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart3789866585
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11649395?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11649395?buildTab=artifacts#/) on master @ [ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a](https://github.com/cockroachdb/cockroach/commits/ad4e53f6b4dda9a76c60d4fe52f6f9d7c6afde2a):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 6218 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:613 +0x68
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0xb0b0870, 0xc0001b4018}, {0xc0053bd640, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:655 +0xd9
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x1f1
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2264 +0x16a
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0x134
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x158
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x93
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xf3
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc003f84680)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xf5
testing.tRunner(0xc003f84680, 0xc0063f6fc0)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart3054665982
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11703119?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11703119?buildTab=artifacts#/) on master @ [46fedddafbff155b72e3853c4f32157cbfebcd35](https://github.com/cockroachdb/cockroach/commits/46fedddafbff155b72e3853c4f32157cbfebcd35):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 1269751 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:615 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x6400448, 0xc000084080}, {0xc002032d78, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:657 +0x8a
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:162 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2264 +0x119
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:252 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:188 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:217 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc002815380)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc002815380, 0xc00d449b40)
	GOROOT/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1629 +0x3ea
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart2318722294
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStart/double_start
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart_double_start1082805071
    test_log_scope.go:81: use -show-logs to present logs inline
=== NAME  TestStart
    test_server_shim.go:126: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11720310?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11720310?buildTab=artifacts#/) on master @ [9959523f444ac2e764d6b2724df35368bc4d0d8a](https://github.com/cockroachdb/cockroach/commits/9959523f444ac2e764d6b2724df35368bc4d0d8a):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 915070 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:615 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x64009e8, 0xc000084080}, {0xc0041fad78, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:657 +0x8a
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:163 +0x15e
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2301 +0x119
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:250 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:186 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:215 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc004bd36c0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc004bd36c0, 0xc000ce0720)
	GOROOT/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1629 +0x3ea
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart1789325919
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStart/double_start
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart_double_start1066813586
    test_log_scope.go:81: use -show-logs to present logs inline
=== NAME  TestStart
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11738408?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11738408?buildTab=artifacts#/) on master @ [b5cef207d5b5905f8a845da5795f1231c7c7545c](https://github.com/cockroachdb/cockroach/commits/b5cef207d5b5905f8a845da5795f1231c7c7545c):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 889895 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:615 +0x36
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0x640d708, 0xc000084080}, {0xc002ba2d58, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:657 +0x8a
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:163 +0x165
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2312 +0x119
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:250 +0xe5
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:186 +0x118
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:215 +0x85
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xaa
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc003fa56c0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xcd
testing.tRunner(0xc003fa56c0, 0xc000cd2340)
	GOROOT/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1629 +0x3ea
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart685369792
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStart/double_start
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart_double_start2862971830
    test_log_scope.go:81: use -show-logs to present logs inline
=== NAME  TestStart
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestStart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11738410?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11738410?buildTab=artifacts#/) on master @ [b5cef207d5b5905f8a845da5795f1231c7c7545c](https://github.com/cockroachdb/cockroach/commits/b5cef207d5b5905f8a845da5795f1231c7c7545c):

Fatal error:

```
panic: attempting to start span with parent from different Tracer.
parent operation: refresh-protectedts-cache, tracer created at:

```
Stack: 

```
goroutine 6649 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracer()
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:615 +0x68
github.com/cockroachdb/cockroach/pkg/util/tracing.NewTracerWithOpt({0xafe6da8, 0xc00013e018}, {0xc003d9f6c8, 0x2, 0x0?})
	github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:657 +0x13d
github.com/cockroachdb/cockroach/pkg/server.makeTestConfigFromParams({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:163 +0x1fd
github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New({}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:2312 +0x17b
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:250 +0x134
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnlyE({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:186 +0x158
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerOnly({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:215 +0x93
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func1()
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:302 +0xf3
pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test.TestStart.func2(0xc0052d31e0)
	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:326 +0xf5
testing.tRunner(0xc0052d31e0, 0xc006862a60)
	GOROOT/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1629 +0x806
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStart
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart1027495074
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStart/double_start
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/38ac07ab4502a5d29d43ab1930ae73cb/logTestStart_double_start287218637
    test_log_scope.go:81: use -show-logs to present logs inline
=== NAME  TestStart
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/protectedts/ptcache/cache_test.go
+++ b/pkg/kv/kvserver/protectedts/ptcache/cache_test.go
@@ -298,7 +298,7 @@ func TestStart(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	setup := func() (serverutils.TestServerInterface, *ptcache.Cache) {
+	setup := func() (serverutils.ApplicationLayerInterface, *ptcache.Cache, func()) {
 		srv := serverutils.StartServerOnly(t,
 			base.TestServerArgs{
 				Knobs: base.TestingKnobs{
@@ -317,25 +317,26 @@ func TestStart(t *testing.T) {
 			DB:       execCfg.InternalDB,
 			Storage:  p,
 		})
-		return srv, c
+		return s, c, func() { srv.Stopper().Stop(ctx) }
 	}
 
 	t.Run("double start", func(t *testing.T) {
 		defer log.Scope(t).Close(t)
 
-		srv, c := setup()
-		defer srv.Stopper().Stop(ctx)
-		require.NoError(t, c.Start(ctx, srv.Stopper()))
-		require.EqualError(t, c.Start(ctx, srv.Stopper()),
+		s, c, cleanup := setup()
+		defer cleanup()
+		require.NoError(t, c.Start(ctx, s.AppStopper()))
+		require.EqualError(t, c.Start(ctx, s.AppStopper()),
 			"cannot start a Cache more than once")
 	})
 
 	t.Run("already stopped", func(t *testing.T) {
 		defer log.Scope(t).Close(t)
 
-		srv, c := setup()
-		srv.Stopper().Stop(ctx)
-		require.EqualError(t, c.Start(ctx, srv.Stopper()),
+		s, c, cleanup := setup()
+		defer cleanup()
+		s.AppStopper().Stop(ctx)
+		require.EqualError(t, c.Start(ctx, s.AppStopper()),
 			stop.ErrUnavailable.Error())
 	})
 }

--- a/pkg/server/tenant.go
+++ b/pkg/server/tenant.go
@@ -1174,9 +1174,8 @@ func makeTenantSQLServerArgs(
 	circularJobRegistry := &jobs.Registry{}
 
 	// Initialize the protectedts subsystem in multi-tenant clusters.
-	var protectedTSProvider protectedts.Provider
 	protectedtsKnobs, _ := baseCfg.TestingKnobs.ProtectedTS.(*protectedts.TestingKnobs)
-	pp, err := ptprovider.New(ptprovider.Config{
+	protectedTSProvider, err := ptprovider.New(ptprovider.Config{
 		DB:       internalExecutorFactory,
 		Settings: st,
 		Knobs:    protectedtsKnobs,
@@ -1192,8 +1191,7 @@ func makeTenantSQLServerArgs(
 	if err != nil {
 		return sqlServerArgs{}, err
 	}
-	registry.AddMetricStruct(pp.Metrics())
-	protectedTSProvider = pp
+	registry.AddMetricStruct(protectedTSProvider.Metrics())
 
 	recorder := status.NewMetricsRecorder(
 		sqlCfg.TenantID, tenantNameContainer, nil /* nodeLiveness */, nil, /* remoteClocks */

